.github/workflows/checks.yml

```diff
@@ -0,0 +1,50 @@
+name: Checks
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  test_and_lint:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache/Restore Mint packages
+        id: mint-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/mint
+          key: ${{ runner.os }}-mint-${{ hashFiles('**/Mintfile') }}
+          restore-keys: ${{ runner.os }}-mint-
+
+      - name: Install package manager "Mint"
+        run: |
+          brew upgrade mint || brew install mint || true
+
+      - name: Install command line tool (if not yet cached)
+        if: steps.mint-cache.outputs.cache-hit != 'true'
+        run: mint bootstrap
+
+      - name: Run SwiftFormat lint
```

Comment:
How does it work? Does it alter the code? 
If so, I'd suggest running just the linter, without formating the code, and returning an error if there are problems with code formatting. The developer should run formatter and commit changes in that scenario.

If I misunderstood something and it works as I wished it was, disregard this comment.

Reply:
No, with with the "--lint" flag it only makes sure all rules are satisflied, without altering the code. I also added a pre-push hook for it